x/ecocredit/client/basket/tx_create.go

```diff
@@ -0,0 +1,205 @@
+					return fmt.Errorf("failed to parse min_start_date: %w", err)
+				}
+				minStartDate, err := types.TimestampProto(minStartDateTime)
+				if err != nil {
+					return fmt.Errorf("failed to parse min_start_date: %w", err)
+				}
+				dateCriteria.Sum = &basket.DateCriteria_MinStartDate{
+					MinStartDate: minStartDate,
+				}
+			}
+
+			if startDateWindowString != "" {
+				startDateWindowInt, err := cmd.Flags().GetInt64(FlagMinimumStartDate)
+				if err != nil {
+					return err
+				}
+				startDateWindowDuration := time.Duration(startDateWindowInt)
+				if err != nil {
+					return fmt.Errorf("failed to parse start-date-window: %w", err)
+				}
+				startDateWindow := types.DurationProto(startDateWindowDuration)
+				if err != nil {
+					return fmt.Errorf("failed to parse start-date-window: %w", err)
+				}
+				dateCriteria.Sum = &basket.DateCriteria_StartDateWindow{
+					StartDateWindow: startDateWindow,
+				}
+			}
+
+			feeString, err := cmd.Flags().GetString(FlagBasketFee)
+			if err != nil {
+				return err
+			}
+			fee, err := sdk.ParseCoinsNormalized(feeString)
+			if err != nil {
+				return fmt.Errorf("failed to parse basket_fee: %w", err)
+			}
+
+			msg := basket.MsgCreate{
+				Curator:           clientCtx.FromAddress.String(),
+				Name:              args[0],
+				DisplayName:       displayName,
+				Exponent:          uint32(exponent),
+				DisableAutoRetire: disableAutoRetire,
+				CreditTypeName:    creditTypeName,
+				AllowedClasses:    allowedClasses,
+				DateCriteria:      &dateCriteria,
+				Fee:               fee,
+			}
+
+			if err := msg.ValidateBasic(); err != nil {
+				return err
+			}
+
+			return tx.GenerateOrBroadcastTxCLI(clientCtx, cmd.Flags(), &msg)
+		},
+	}
+
+	flags.AddTxFlagsToCmd(cmd)
+
+	// command flags
+	cmd.Flags().String(FlagDisplayName, "", "the name used to create a bank denom display name")
+	cmd.Flags().String(FlagExponent, "", "the exponent used for converting credits to basket tokens")
+	cmd.Flags().Bool(FlagDisableAutoRetire, false, "dictates whether credits will be auto-retired upon taking")
+	cmd.Flags().String(FlagCreditTypeName, "", "filters against credits from this credit type name (e.g. \"carbon\")")
+	cmd.Flags().String(FlagAllowedClasses, "", "comma separated (no spaces) list of credit classes allowed to be put in the basket (e.g. \"C01,C02\")")
+	cmd.Flags().String(FlagMinimumStartDate, "", "the earliest start date for batches of credits allowed into the basket (e.g. \"2012-01-01\")")
+	cmd.Flags().String(FlagStartDateWindow, "", "sets a cutoff for batch start dates when adding new credits to the basket (e.g. \"1325404800\")")
+	cmd.Flags().String(FlagBasketFee, "", "the fee that the curator will pay to create the basket (e.g. \"20regen\")")
+
+	// required flags
+	cmd.MarkFlagRequired(FlagDisplayName)
+	cmd.MarkFlagRequired(FlagExponent)
+	cmd.MarkFlagRequired(FlagCreditTypeName)
+	cmd.MarkFlagRequired(FlagAllowedClasses)
+	cmd.MarkFlagRequired(FlagBasketFee)
```

Comment:
is fee actually required? iirc there was a discussion around empty fees in the validate basic for MsgCreate being ok

Reply:
fee is not required